sandbox/cgroup/tracking_test.go

```diff
@@ -0,0 +1,580 @@
+		return nil, fmt.Errorf("system bus is not available for testing")
+	}
+	restore := dbusutil.MockConnections(noSystemBus, dbustest.StubConnection)
+	defer restore()
+
+	// Pretend we are a root user so that we attempt to use the system bus as fallback.
+	restore = cgroup.MockOsGetuid(0)
+	defer restore()
+	// Pretend our PID is this value.
+	restore = cgroup.MockOsGetpid(312123)
+	defer restore()
+
+	// Rig the random UUID generator to return this value.
+	uuid := "cc98cd01-6a25-46bd-b71b-82069b71b770"
+	restore = cgroup.MockRandomUUID(uuid)
+	defer restore()
+
+	// Calling StartTransientUnit fails so that we try to use the system bus as fallback.
+	restore = cgroup.MockDoCreateTransientScope(func(conn *dbus.Conn, unitName string, pid int) error {
+		return cgroup.ErrDBusSpawnChildExited
+	})
+	defer restore()
+
+	// Disable the cgroup analyzer function as we don't expect it to be used in this test.
+	restore = cgroup.MockCgroupProcessPathInTrackingCgroup(func(pid int) (string, error) {
+		panic("we are not expecting this call")
+	})
+	defer restore()
+
+	// Attempts to create a transient scope fail with a special error
+	// indicating that we cannot track application process.
+	err := cgroup.CreateTransientScopeForTracking("snap.pkg.app")
+	c.Assert(err, ErrorMatches, "cannot track application process")
+}
+
+func (s *trackingSuite) TestCreateTransientScopeForTrackingUnhappyNoDBus(c *C) {
+	// Pretend that refresh app awareness is enabled
+	enableFeatures(c, features.RefreshAppAwareness)
+
+	// Make it appear that DBus is entirely unavailable.
+	noBus := func() (*dbus.Conn, error) {
+		return nil, fmt.Errorf("dbus is not available for testing")
+	}
+	restore := dbusutil.MockConnections(noBus, noBus)
+	defer restore()
+
+	// Pretend we are a root user so that we attempt to use the system bus as fallback.
+	restore = cgroup.MockOsGetuid(0)
+	defer restore()
+	// Pretend our PID is this value.
+	restore = cgroup.MockOsGetpid(312123)
+	defer restore()
+
+	// Rig the random UUID generator to return this value.
+	uuid := "cc98cd01-6a25-46bd-b71b-82069b71b770"
+	restore = cgroup.MockRandomUUID(uuid)
+	defer restore()
+
+	// Calling StartTransientUnit is not attempted without a DBus connection.
+	restore = cgroup.MockDoCreateTransientScope(func(conn *dbus.Conn, unitName string, pid int) error {
+		panic("we are not expecting this call")
+	})
+	defer restore()
+
+	// Disable the cgroup analyzer function as we don't expect it to be used in this test.
+	restore = cgroup.MockCgroupProcessPathInTrackingCgroup(func(pid int) (string, error) {
+		panic("we are not expecting this call")
+	})
+	defer restore()
+
+	// Attempts to create a transient scope fail with a special error
+	// indicating that we cannot track application process.
+	err := cgroup.CreateTransientScopeForTracking("snap.pkg.app")
+	c.Assert(err, ErrorMatches, "cannot track application process")
+}
+
+func (s *trackingSuite) TestCreateTransientScopeForTrackingSilentlyFails(c *C) {
```

Comment:
thanks for this test, also this makes more clear my questions about the code above, the verification is really checking that the kernel bug didn't happen, so if the kernel bug happens we fail expectedly

Reply:
It's not really a bug. It's a configuration issue. The kernel is conservative and prevents users from stealing processes or dumping their resources onto other users. I've revised the comment above to make this more clear.